codespell_lib/_codespell.py

```diff
@@ -32,6 +32,8 @@
 # endpoint.  Emails are more restrictive, so the endpoint remains flexible.
 uri_regex_def = (u"(\\b(?:https?|[ts]?ftp|file|git|smb)://[^\\s]+(?=$|\\s)|"
                  u"\\b[\\w.%+-]+@[\\w.-]+\\b)")
+inline_ignore_regex = re.compile(
+    r"[^\w\s]\s?codespell:ignore\b(\s+(?P<words>[\w,]*))?")
```

Comment:
To make this compatible with https://github.com/codespell-project/codespell/issues/1212#issuecomment-1181813266 (i.e. https://cspell.org/configuration/document-settings/), how about

```suggestion
    r"[^\w\s]\s?(?:codespell|cspell|spell-checker):ignore\b(\s+(?P<words>[\w,]*))?", flags=re.I)
```

Reply:
`cspell` looks like a bad idea for the reasons explained in https://github.com/codespell-project/codespell/issues/1212#issuecomment-1193002440. I would also defer `spell-checker` to a later time when in document settings consistent with _cspell_ are implemented.